src/types/resolveDescriptors.ts

```diff
@@ -1552,6 +1550,18 @@ export function resolveDescriptors(ctx: CompilerContext, Ast: FactoryAst) {
                         `Trait "${tr.name}" requires field "${f.name}" of type "${printTypeRef(f.type)}"`,
                         t.ast.loc,
                     );
+                } else if (
+                    f.as !== ex.as &&
+                    f.as !== "int257" &&
+                    ex.as !== "int257"
```

Comment:
I think something like

```tact
trait Trait {
    i: Int as int257;
}

contract Foo with Trait {
    i: Int as coins = 0;
}
```

won't be reported

Reply:
Indeed, I simplified the conditions too much, the new ones should work fine